packages/uui-button/lib/uui-button.element.ts

```diff
@@ -427,9 +436,12 @@ export class UUIButtonElement extends FormControlMixin(
   @query('#button')
   protected _button!: HTMLInputElement;
 
+  #popoverIsOpen = false;
+
   constructor() {
     super();
     this.addEventListener('click', this._onHostClick);
+    this.addEventListener('uui-popover-before-toggle', this.#popoverListener);
```

Comment:
Let's turn this into an Event class, like many of our other events. So we can have types for the details, plus it gives confirmation that the string is correct. :-)